### DISCUSSION TITLE
GITHUB_SHA not always correct

### DISCUSSION THREAD

**Discussion by Charles-Gagnon** [community/community#25341] [Actions]
<p>Sometimes the GITHUB_SHA doesn’t seem to match the commits for the PR branch. This doesn’t happen all the time - but when it does happen it seems to be persistent for that PR no matter how many commits you push to the branch being merged.</p>
<p>example : <a href="https://github.com/microsoft/azuredatastudio/pull/10879/checks?check_run_id=766753586" rel="nofollow noopener">https://github.com/microsoft/azuredatastudio/pull/10879/checks?check_run_id=766753586</a></p>
<p>Locally I run these commands to switch to the PR branch :</p>
<p>git fetch origin +refs/pull/10879/merge:refs/remotes/pull/10879/merge<br>
git checkout refs/remotes/pull/10879/merge</p>
<p>And then git log :</p>
<p>d3cf158c11fa584702248cd2e54f84a44166f7e9 (HEAD, pull/10879/merge) Merge e17cf53bd3b48e03fc4d86edb93107f5c9e4bb94 into 233646330e08351de40267c2c26faf644ac05155</p>
<p>In the failed check you can see that Coveralls upload task failed with this error :</p>
<p>##[error]Error: Command failed: git cat-file -p dae9cdfaf4d02ebd1900c62f150297dd3b0553e2</p>
<p>The hash it uses is the GITHUB_SHA environment variable set for the action. But that doesn’t match the hash of the merge commit.</p>
<p>This works fine in other cases though. Example : <a href="https://github.com/microsoft/azuredatastudio/pull/10855/checks?check_run_id=766338968" rel="nofollow noopener">https://github.com/microsoft/azuredatastudio/pull/10855/checks?check_run_id=766338968</a></p>
<p>I did the same thing there - locally ran these commands :</p>
<p>git fetch origin +refs/pull/10855/merge:refs/remotes/pull/10855/merge<br>
git checkout refs/remotes/pull/10855/merge<br>
git log --pretty=oneline</p>
<p>and got this as the merge commit</p>
<p>e4b1ab3b1f9a231ad98eefe0b0e39401e7521499 (HEAD, pull/10855/merge) Merge f2628d1d2340fb69880279bebfe91f1c50457bb1 into a7110d89160e25cf914030dd467f3d85c0292692</p>
<p>which matches the GITHUB_SHA indicated in the action</p>
<p>GITHUB_SHA=e4b1ab3b1f9a231ad98eefe0b0e39401e7521499</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/charles-gagnon">@Charles-Gagnon</a>,</p>
<p>Could you please add below script to confirm the github sha value? I forked the repo but failed to repro on my side.</p>
<pre><code class="lang-auto">      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
</code></pre>
<p>Thanks.</p>

**Comment by Charles-Gagnon** [comment]
<aside class="onebox githubpullrequest">
  <header class="source">
      <a href="https://github.com/microsoft/azuredatastudio/pull/10879/checks?check_run_id=773856282" target="_blank" rel="nofollow noopener">github.com/microsoft/azuredatastudio</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/microsoft/azuredatastudio/pull/10879" target="_blank" rel="nofollow noopener">Initial unit tests for class NotebookEditor</a>
    </h4>

    <div class="branches">
      <code>microsoft:main</code> ← <code>microsoft:feat/nbCover</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-06-11" data-time="19:16:46" data-timezone="UTC">07:16PM - 11 Jun 20 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/ranasaria" target="_blank" rel="nofollow noopener">
          <img alt="ranasaria" src="https://user-images.githubusercontent.com/28519865/181090517-0aff924a-ab5f-4b21-b2e3-86067ff7542b.png" class="onebox-avatar-inline" width="20" height="20">
          ranasaria
        </a>
      </div>

      <div class="lines" title="21 commits changed 7 files with 245 additions and 56 deletions">
        <a href="https://github.com/microsoft/azuredatastudio/pull/10879/files" target="_blank" rel="nofollow noopener">
          <span class="added">+245</span>
          <span class="removed">-56</span>
        </a>
      </div>
    </div>

  </div>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>sha from the context :</p>
<p><code>  "sha": "49b472bccdca381a7d8b92761fbf037a4af78800",</code></p>
<p>from the checkout step :</p>
<p><code>HEAD is now at d631d0b28 Merge 95caa894062bb5ec9cb7a88e1f1a8793d665e7b8 into d9e70731f49bfdac9f6b5ae11a8c2cfd4c95a893</code></p>
<p>I get the same thing if I run this command locally</p>
<p><code>git checkout --progress --force refs/remotes/pull/10879/merge</code></p>
<pre><code class="lang-auto">GITHUB_CONTEXT: {
  "token": "***",
  "job": "linux",
  "ref": "refs/pull/10879/merge",
  "sha": "49b472bccdca381a7d8b92761fbf037a4af78800",
  "repository": "microsoft/azuredatastudio",
  "repository_owner": "microsoft",
  "repositoryUrl": "git://github.com/microsoft/azuredatastudio.git",
  "run_id": "136329483",
  "run_number": "4387",
  "actor": "Charles-Gagnon",
  "workflow": "CI",
  "head_ref": "feat/nbCover",
  "base_ref": "main",
  "event_name": "pull_request",
  "event": {
    "action": "synchronize",
    "after": "95caa894062bb5ec9cb7a88e1f1a8793d665e7b8",
    "before": "c3d12644dea877e05dc01a66067352a24aea0602",
    "enterprise": {
      "avatar_url": "https://avatars0.githubusercontent.com/b/1578?v=4",
      "created_at": "2019-12-09T02:41:53Z",
      "description": "Microsoft's organizations for open source collaboration",
      "html_url": "https://github.com/enterprises/microsoftopensource",
      "id": 1578,
      "name": "Microsoft Open Source",
      "node_id": "MDEwOkVudGVycHJpc2UxNTc4",
      "slug": "microsoftopensource",
      "updated_at": "2020-05-19T18:21:45Z",
      "website_url": "https://opensource.microsoft.com"
    },
    "number": 10879,
    "organization": {
      "avatar_url": "https://avatars2.githubusercontent.com/u/6154722?v=4",
      "description": "Open source, from Microsoft with love",
      "events_url": "https://api.github.com/orgs/microsoft/events",
      "hooks_url": "https://api.github.com/orgs/microsoft/hooks",
      "id": 6154722,
      "issues_url": "https://api.github.com/orgs/microsoft/issues",
      "login": "microsoft",
      "members_url": "https://api.github.com/orgs/microsoft/members{/member}",
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjYxNTQ3MjI=",
      "public_members_url": "https://api.github.com/orgs/microsoft/public_members{/member}",
      "repos_url": "https://api.github.com/orgs/microsoft/repos",
      "url": "https://api.github.com/orgs/microsoft"
    },
    "pull_request": {
      "_links": {
        "comments": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/issues/10879/comments"
        },
        "commits": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/pulls/10879/commits"
        },
        "html": {
          "href": "https://github.com/microsoft/azuredatastudio/pull/10879"
        },
        "issue": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/issues/10879"
        },
        "review_comment": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/pulls/comments{/number}"
        },
        "review_comments": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/pulls/10879/comments"
        },
        "self": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/pulls/10879"
        },
        "statuses": {
          "href": "https://api.github.com/repos/microsoft/azuredatastudio/statuses/95caa894062bb5ec9cb7a88e1f1a8793d665e7b8"
        }
      },
      "active_lock_reason": null,
      "additions": 245,
      "assignee": {
        "avatar_url": "https://avatars2.githubusercontent.com/u/41588310?v=4",
        "events_url": "https://api.github.com/users/ranasaria/events{/privacy}",
        "followers_url": "https://api.github.com/users/ranasaria/followers",
        "following_url": "https://api.github.com/users/ranasaria/following{/other_user}",
        "gists_url": "https://api.github.com/users/ranasaria/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/ranasaria",
        "id": 41588310,
        "login": "ranasaria",
        "node_id": "MDQ6VXNlcjQxNTg4MzEw",
        "organizations_url": "https://api.github.com/users/ranasaria/orgs",
        "received_events_url": "https://api.github.com/users/ranasaria/received_events",
        "repos_url": "https://api.github.com/users/ranasaria/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/ranasaria/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/ranasaria/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/ranasaria"
      },
      "assignees": [
        {
          "avatar_url": "https://avatars2.githubusercontent.com/u/41588310?v=4",
          "events_url": "https://api.github.com/users/ranasaria/events{/privacy}",
          "followers_url": "https://api.github.com/users/ranasaria/followers",
          "following_url": "https://api.github.com/users/ranasaria/following{/other_user}",
          "gists_url": "https://api.github.com/users/ranasaria/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/ranasaria",
          "id": 41588310,
          "login": "ranasaria",
          "node_id": "MDQ6VXNlcjQxNTg4MzEw",
          "organizations_url": "https://api.github.com/users/ranasaria/orgs",
          "received_events_url": "https://api.github.com/users/ranasaria/received_events",
          "repos_url": "https://api.github.com/users/ranasaria/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/ranasaria/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/ranasaria/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/ranasaria"
        }
      ],
      "author_association": "CONTRIBUTOR",
      "base": {
        "label": "microsoft:main",
        "ref": "main",
        "repo": {
          "archive_url": "https://api.github.com/repos/microsoft/azuredatastudio/{archive_format}{/ref}",
          "archived": false,
          "assignees_url": "https://api.github.com/repos/microsoft/azuredatastudio/assignees{/user}",
          "blobs_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/blobs{/sha}",
          "branches_url": "https://api.github.com/repos/microsoft/azuredatastudio/branches{/branch}",
          "clone_url": "https://github.com/microsoft/azuredatastudio.git",
          "collaborators_url": "https://api.github.com/repos/microsoft/azuredatastudio/collaborators{/collaborator}",
          "comments_url": "https://api.github.com/repos/microsoft/azuredatastudio/comments{/number}",
          "commits_url": "https://api.github.com/repos/microsoft/azuredatastudio/commits{/sha}",
          "compare_url": "https://api.github.com/repos/microsoft/azuredatastudio/compare/{base}...{head}",
          "contents_url": "https://api.github.com/repos/microsoft/azuredatastudio/contents/{+path}",
          "contributors_url": "https://api.github.com/repos/microsoft/azuredatastudio/contributors",
          "created_at": "2017-11-02T01:00:13Z",
          "default_branch": "main",
          "deployments_url": "https://api.github.com/repos/microsoft/azuredatastudio/deployments",
          "description": "Azure Data Studio is a data management tool that enables working with SQL Server, Azure SQL DB and SQL DW from Windows, macOS and Linux.",
          "disabled": false,
          "downloads_url": "https://api.github.com/repos/microsoft/azuredatastudio/downloads",
          "events_url": "https://api.github.com/repos/microsoft/azuredatastudio/events",
          "fork": false,
          "forks": 522,
          "forks_count": 522,
          "forks_url": "https://api.github.com/repos/microsoft/azuredatastudio/forks",
          "full_name": "microsoft/azuredatastudio",
          "git_commits_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/commits{/sha}",
          "git_refs_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/refs{/sha}",
          "git_tags_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/tags{/sha}",
          "git_url": "git://github.com/microsoft/azuredatastudio.git",
          "has_downloads": true,
          "has_issues": true,
          "has_pages": false,
          "has_projects": true,
          "has_wiki": true,
          "homepage": "",
          "hooks_url": "https://api.github.com/repos/microsoft/azuredatastudio/hooks",
          "html_url": "https://github.com/microsoft/azuredatastudio",
          "id": 109200851,
          "issue_comment_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/comments{/number}",
          "issue_events_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/events{/number}",
          "issues_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues{/number}",
          "keys_url": "https://api.github.com/repos/microsoft/azuredatastudio/keys{/key_id}",
          "labels_url": "https://api.github.com/repos/microsoft/azuredatastudio/labels{/name}",
          "language": "TypeScript",
          "languages_url": "https://api.github.com/repos/microsoft/azuredatastudio/languages",
          "license": {
            "key": "other",
            "name": "Other",
            "node_id": "MDc6TGljZW5zZTA=",
            "spdx_id": "NOASSERTION",
            "url": null
          },
          "merges_url": "https://api.github.com/repos/microsoft/azuredatastudio/merges",
          "milestones_url": "https://api.github.com/repos/microsoft/azuredatastudio/milestones{/number}",
          "mirror_url": null,
          "name": "azuredatastudio",
          "node_id": "MDEwOlJlcG9zaXRvcnkxMDkyMDA4NTE=",
          "notifications_url": "https://api.github.com/repos/microsoft/azuredatastudio/notifications{?since,all,participating}",
          "open_issues": 1776,
          "open_issues_count": 1776,
          "owner": {
            "avatar_url": "https://avatars2.githubusercontent.com/u/6154722?v=4",
            "events_url": "https://api.github.com/users/microsoft/events{/privacy}",
            "followers_url": "https://api.github.com/users/microsoft/followers",
            "following_url": "https://api.github.com/users/microsoft/following{/other_user}",
            "gists_url": "https://api.github.com/users/microsoft/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/microsoft",
            "id": 6154722,
            "login": "microsoft",
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjYxNTQ3MjI=",
            "organizations_url": "https://api.github.com/users/microsoft/orgs",
            "received_events_url": "https://api.github.com/users/microsoft/received_events",
            "repos_url": "https://api.github.com/users/microsoft/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/microsoft/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/microsoft/subscriptions",
            "type": "Organization",
            "url": "https://api.github.com/users/microsoft"
          },
          "private": false,
          "pulls_url": "https://api.github.com/repos/microsoft/azuredatastudio/pulls{/number}",
          "pushed_at": "2020-06-15T18:46:53Z",
          "releases_url": "https://api.github.com/repos/microsoft/azuredatastudio/releases{/id}",
          "size": 93036,
          "ssh_url": "git@github.com:microsoft/azuredatastudio.git",
          "stargazers_count": 5773,
          "stargazers_url": "https://api.github.com/repos/microsoft/azuredatastudio/stargazers",
          "statuses_url": "https://api.github.com/repos/microsoft/azuredatastudio/statuses/{sha}",
          "subscribers_url": "https://api.github.com/repos/microsoft/azuredatastudio/subscribers",
          "subscription_url": "https://api.github.com/repos/microsoft/azuredatastudio/subscription",
          "svn_url": "https://github.com/microsoft/azuredatastudio",
          "tags_url": "https://api.github.com/repos/microsoft/azuredatastudio/tags",
          "teams_url": "https://api.github.com/repos/microsoft/azuredatastudio/teams",
          "trees_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/trees{/sha}",
          "updated_at": "2020-06-15T18:25:36Z",
          "url": "https://api.github.com/repos/microsoft/azuredatastudio",
          "watchers": 5773,
          "watchers_count": 5773
        },
        "sha": "d9e70731f49bfdac9f6b5ae11a8c2cfd4c95a893",
        "user": {
          "avatar_url": "https://avatars2.githubusercontent.com/u/6154722?v=4",
          "events_url": "https://api.github.com/users/microsoft/events{/privacy}",
          "followers_url": "https://api.github.com/users/microsoft/followers",
          "following_url": "https://api.github.com/users/microsoft/following{/other_user}",
          "gists_url": "https://api.github.com/users/microsoft/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/microsoft",
          "id": 6154722,
          "login": "microsoft",
          "node_id": "MDEyOk9yZ2FuaXphdGlvbjYxNTQ3MjI=",
          "organizations_url": "https://api.github.com/users/microsoft/orgs",
          "received_events_url": "https://api.github.com/users/microsoft/received_events",
          "repos_url": "https://api.github.com/users/microsoft/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/microsoft/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/microsoft/subscriptions",
          "type": "Organization",
          "url": "https://api.github.com/users/microsoft"
        }
      },
      "body": "&lt;!-- Thank you for submitting a Pull Request. Please:\r\n* Read our Pull Request guidelines:\r\n  https://github.com/Microsoft/azuredatastudio/wiki/How-to-Contribute#pull-requests.\r\n* Associate an issue with the Pull Request.\r\n* Ensure that the code is up-to-date with the `master` branch.\r\n* Include a description of the proposed changes and how to test them.\r\n--&gt;\r\nThis PR checks in the initial few unit tests for NotebookEditor class. It also fixes a small stack overflow issue in promise.ts and does some spelling fixes and readability renames.\r\nThis PR fixes #10875 \r\n\r\nTested all unit tests on a dev box. The following is the coverage added to the NotebookEditor class(from basically 0) from these tests:\r\n![image](https://user-images.githubusercontent.com/41588310/84447547-225fca00-abfd-11ea-9e8c-9a42a4eca7e7.png)\r\n",
      "changed_files": 7,
      "closed_at": null,
      "comments": 1,
      "comments_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/10879/comments",
      "commits": 21,
      "commits_url": "https://api.github.com/repos/microsoft/azuredatastudio/pulls/10879/commits",
      "created_at": "2020-06-11T19:16:46Z",
      "deletions": 56,
      "diff_url": "https://github.com/microsoft/azuredatastudio/pull/10879.diff",
      "draft": false,
      "head": {
        "label": "microsoft:feat/nbCover",
        "ref": "feat/nbCover",
        "repo": {
          "archive_url": "https://api.github.com/repos/microsoft/azuredatastudio/{archive_format}{/ref}",
          "archived": false,
          "assignees_url": "https://api.github.com/repos/microsoft/azuredatastudio/assignees{/user}",
          "blobs_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/blobs{/sha}",
          "branches_url": "https://api.github.com/repos/microsoft/azuredatastudio/branches{/branch}",
          "clone_url": "https://github.com/microsoft/azuredatastudio.git",
          "collaborators_url": "https://api.github.com/repos/microsoft/azuredatastudio/collaborators{/collaborator}",
          "comments_url": "https://api.github.com/repos/microsoft/azuredatastudio/comments{/number}",
          "commits_url": "https://api.github.com/repos/microsoft/azuredatastudio/commits{/sha}",
          "compare_url": "https://api.github.com/repos/microsoft/azuredatastudio/compare/{base}...{head}",
          "contents_url": "https://api.github.com/repos/microsoft/azuredatastudio/contents/{+path}",
          "contributors_url": "https://api.github.com/repos/microsoft/azuredatastudio/contributors",
          "created_at": "2017-11-02T01:00:13Z",
          "default_branch": "main",
          "deployments_url": "https://api.github.com/repos/microsoft/azuredatastudio/deployments",
          "description": "Azure Data Studio is a data management tool that enables working with SQL Server, Azure SQL DB and SQL DW from Windows, macOS and Linux.",
          "disabled": false,
          "downloads_url": "https://api.github.com/repos/microsoft/azuredatastudio/downloads",
          "events_url": "https://api.github.com/repos/microsoft/azuredatastudio/events",
          "fork": false,
          "forks": 522,
          "forks_count": 522,
          "forks_url": "https://api.github.com/repos/microsoft/azuredatastudio/forks",
          "full_name": "microsoft/azuredatastudio",
          "git_commits_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/commits{/sha}",
          "git_refs_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/refs{/sha}",
          "git_tags_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/tags{/sha}",
          "git_url": "git://github.com/microsoft/azuredatastudio.git",
          "has_downloads": true,
          "has_issues": true,
          "has_pages": false,
          "has_projects": true,
          "has_wiki": true,
          "homepage": "",
          "hooks_url": "https://api.github.com/repos/microsoft/azuredatastudio/hooks",
          "html_url": "https://github.com/microsoft/azuredatastudio",
          "id": 109200851,
          "issue_comment_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/comments{/number}",
          "issue_events_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/events{/number}",
          "issues_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues{/number}",
          "keys_url": "https://api.github.com/repos/microsoft/azuredatastudio/keys{/key_id}",
          "labels_url": "https://api.github.com/repos/microsoft/azuredatastudio/labels{/name}",
          "language": "TypeScript",
          "languages_url": "https://api.github.com/repos/microsoft/azuredatastudio/languages",
          "license": {
            "key": "other",
            "name": "Other",
            "node_id": "MDc6TGljZW5zZTA=",
            "spdx_id": "NOASSERTION",
            "url": null
          },
          "merges_url": "https://api.github.com/repos/microsoft/azuredatastudio/merges",
          "milestones_url": "https://api.github.com/repos/microsoft/azuredatastudio/milestones{/number}",
          "mirror_url": null,
          "name": "azuredatastudio",
          "node_id": "MDEwOlJlcG9zaXRvcnkxMDkyMDA4NTE=",
          "notifications_url": "https://api.github.com/repos/microsoft/azuredatastudio/notifications{?since,all,participating}",
          "open_issues": 1776,
          "open_issues_count": 1776,
          "owner": {
            "avatar_url": "https://avatars2.githubusercontent.com/u/6154722?v=4",
            "events_url": "https://api.github.com/users/microsoft/events{/privacy}",
            "followers_url": "https://api.github.com/users/microsoft/followers",
            "following_url": "https://api.github.com/users/microsoft/following{/other_user}",
            "gists_url": "https://api.github.com/users/microsoft/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/microsoft",
            "id": 6154722,
            "login": "microsoft",
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjYxNTQ3MjI=",
            "organizations_url": "https://api.github.com/users/microsoft/orgs",
            "received_events_url": "https://api.github.com/users/microsoft/received_events",
            "repos_url": "https://api.github.com/users/microsoft/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/microsoft/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/microsoft/subscriptions",
            "type": "Organization",
            "url": "https://api.github.com/users/microsoft"
          },
          "private": false,
          "pulls_url": "https://api.github.com/repos/microsoft/azuredatastudio/pulls{/number}",
          "pushed_at": "2020-06-15T18:46:53Z",
          "releases_url": "https://api.github.com/repos/microsoft/azuredatastudio/releases{/id}",
          "size": 93036,
          "ssh_url": "git@github.com:microsoft/azuredatastudio.git",
          "stargazers_count": 5773,
          "stargazers_url": "https://api.github.com/repos/microsoft/azuredatastudio/stargazers",
          "statuses_url": "https://api.github.com/repos/microsoft/azuredatastudio/statuses/{sha}",
          "subscribers_url": "https://api.github.com/repos/microsoft/azuredatastudio/subscribers",
          "subscription_url": "https://api.github.com/repos/microsoft/azuredatastudio/subscription",
          "svn_url": "https://github.com/microsoft/azuredatastudio",
          "tags_url": "https://api.github.com/repos/microsoft/azuredatastudio/tags",
          "teams_url": "https://api.github.com/repos/microsoft/azuredatastudio/teams",
          "trees_url": "https://api.github.com/repos/microsoft/azuredatastudio/git/trees{/sha}",
          "updated_at": "2020-06-15T18:25:36Z",
          "url": "https://api.github.com/repos/microsoft/azuredatastudio",
          "watchers": 5773,
          "watchers_count": 5773
        },
        "sha": "95caa894062bb5ec9cb7a88e1f1a8793d665e7b8",
        "user": {
          "avatar_url": "https://avatars2.githubusercontent.com/u/6154722?v=4",
          "events_url": "https://api.github.com/users/microsoft/events{/privacy}",
          "followers_url": "https://api.github.com/users/microsoft/followers",
          "following_url": "https://api.github.com/users/microsoft/following{/other_user}",
          "gists_url": "https://api.github.com/users/microsoft/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/microsoft",
          "id": 6154722,
          "login": "microsoft",
          "node_id": "MDEyOk9yZ2FuaXphdGlvbjYxNTQ3MjI=",
          "organizations_url": "https://api.github.com/users/microsoft/orgs",
          "received_events_url": "https://api.github.com/users/microsoft/received_events",
          "repos_url": "https://api.github.com/users/microsoft/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/microsoft/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/microsoft/subscriptions",
          "type": "Organization",
          "url": "https://api.github.com/users/microsoft"
        }
      },
      "html_url": "https://github.com/microsoft/azuredatastudio/pull/10879",
      "id": 433280627,
      "issue_url": "https://api.github.com/repos/microsoft/azuredatastudio/issues/10879",
      "labels": [],
      "locked": false,
      "maintainer_can_modify": false,
      "merge_commit_sha": "a8f0dca87d70de08043bff7db4cadc1314e4bb74",
      "mergeable": null,
      "mergeable_state": "unknown",
      "merged": false,
      "merged_at": null,
      "merged_by": null,
      "milestone": null,
      "node_id": "MDExOlB1bGxSZXF1ZXN0NDMzMjgwNjI3",
      "number": 10879,
      "patch_url": "https://github.com/microsoft/azuredatastudio/pull/10879.patch",
      "rebaseable": null,
...
</code></pre>
<p>(let me know if you need more - size limits mean I can’t paste the whole context)</p>

**Comment by Charles-Gagnon** [comment, marked as answer]
<p>After some help with the investigation it ended up being that the checkout action being used wasn’t pulling the correct version down and so there was a mismatch between what was checked out and what the GITHUB_SHA was.</p>
<p>Updating to v2.2.0 of the checkout action should solve the issue.</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/coverallsapp/github-action/issues/55" target="_blank" rel="nofollow noopener">github.com/coverallsapp/github-action</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/coverallsapp/github-action/issues/55" target="_blank" rel="nofollow noopener">Upload failing randomly during getOptions</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-06-12" data-time="17:40:19" data-timezone="UTC">05:40PM - 12 Jun 20 UTC</span>
      </div>

        <div class="date">
          closed <span class="discourse-local-date" data-format="ll" data-date="2020-06-16" data-time="18:32:14" data-timezone="UTC">06:32PM - 16 Jun 20 UTC</span>
        </div>

      <div class="user">
        <a href="https://github.com/Charles-Gagnon" target="_blank" rel="nofollow noopener">
          <img alt="Charles-Gagnon" src="https://user-images.githubusercontent.com/28519865/181090524-c9df381a-c909-4b1f-acbe-02bd286fefed.jpeg" class="onebox-avatar-inline" width="20" height="20">
          Charles-Gagnon
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">Most of the time the upload task works fine, but occasionally it'll fail with the following error :
[error] "2020-06-12T05:29:37.644Z" 'error from...</p>
</div>

<div class="labels">
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
